packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.js

```diff
@@ -578,8 +578,45 @@ export let CreateTearsheet = forwardRef(
       );
     }, []);
 
+    // track scrolling/intersection of create sections so that we know
+    // which section is active (updates the SideNavItems `isActive` prop)
+    useEffect(() => {
+      if (shouldViewAll) {
+        const tearsheetMainContent = document.querySelector(
+          `.${pkg.prefix}--tearsheet__content`
+        );
+        let options = {
+          root: tearsheetMainContent,
+          rootMargin: '0px',
+          threshold: 0,
+        };
+        const viewAllSections = document.querySelectorAll(
+          `.${pkg.prefix}--tearsheet-create__section.${pkg.prefix}--tearsheet-create__step--visible-section`
+        );
```

Comment:
Just a minor point, but I think it might make better sense to convert the nodelist to an array once, here, rather than converting it every time an intersection observer fires.

```suggestion
        const viewAllSections = Array.from(document.querySelectorAll(
          `.${pkg.prefix}--tearsheet-create__section.${pkg.prefix}--tearsheet-create__step--visible-section`
        ));
```

Reply:
Yeah good idea!

Reply:
Updated